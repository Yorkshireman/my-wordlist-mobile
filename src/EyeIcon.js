import { TextInput } from 'react-native-paper';

export const EyeIcon = (onPress, passwordVisible) => (
  <TextInput.Icon 
    icon={passwordVisible ? 'eye' : 'eye-off'}
    onPress={onPress}
  />
);
