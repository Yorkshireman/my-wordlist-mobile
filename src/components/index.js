export { CreateWordlistEntryForm } from './CreateWordlistEntryForm';
export { Error } from './Error';
export { EyeIcon } from './EyeIcon';
export { Loading } from './Loading';
export { NavigationBar } from './NavigationBar';
export { Wordlist } from './Wordlist';
