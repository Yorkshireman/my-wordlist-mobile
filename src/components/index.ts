export { AddCategoriesForm } from './AddCategoriesForm';
export { ClearIcon } from './ClearIcon';
export { CreateWordlistEntryForm } from './CreateWordlistEntryForm';
export { EditWordForm } from './EditWordForm';
export { Error } from './Error';
export { ExampleSentences } from './ExampleSentences';
export { EyeIcon } from './EyeIcon';
export { Filters } from './Filters';
export { GenerateExampleSentencesScreenWrapper } from './GenerateExampleSentencesScreenWrapper';
export { GenerateExampleSentencesOptionsProvider } from './GenerateExampleSentencesOptionsProvider';
export { Loading } from './Loading';
export { NavigationBar } from './NavigationBar';
export { NotificationProvider } from './NotificationProvider';
export { SentencesGeneratorOptions } from './SentencesGeneratorOptions';
export { Wordlist } from './Wordlist';
