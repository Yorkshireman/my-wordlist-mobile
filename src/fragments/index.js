import { WORDLIST_ENTRY } from './wordlistEntry';

export { WORDLIST_ENTRY };
