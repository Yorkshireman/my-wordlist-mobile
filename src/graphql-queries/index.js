import { MY_WORDLIST } from './myWordlist';
import { MY_WORDLIST_CREATE } from './myWordlistCreate';
import { WORDLIST_ENTRY_CREATE } from './wordlistEntryCreate';
import { WORDLIST_ENTRY_DELETE } from './wordlistEntryDelete';
import { WORDLIST_ENTRY_UPDATE } from './wordlistEntryUpdate';

export {
  MY_WORDLIST,
  MY_WORDLIST_CREATE,
  WORDLIST_ENTRY_CREATE,
  WORDLIST_ENTRY_DELETE,
  WORDLIST_ENTRY_UPDATE
};
