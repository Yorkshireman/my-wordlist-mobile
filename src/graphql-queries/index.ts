export { FETCH_OR_CREATE_EXAMPLE_SENTENCES } from './fetchOrCreateExampleSentences';
export { MY_WORDLIST } from './myWordlist';
export { MY_WORDLIST_CATEGORIES } from './myWordlistCategories';
export { MY_WORDLIST_CREATE } from './myWordlistCreate';
export { WORDLIST_ENTRIES_CREATE } from './wordlistEntriesCreate';
export { WORDLIST_ENTRY_DELETE } from './wordlistEntryDelete';
export { WORDLIST_ENTRY_UPDATE } from './wordlistEntryUpdate';
