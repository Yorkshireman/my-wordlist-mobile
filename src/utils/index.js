export { calculateLongestWordLength } from './calculateLongestWordLength';
export { customColours } from './customColours';
export { isValidWordlistEntry } from './isValidWordlistEntry';
export { parseCategories } from './parseCategories';
export { storeAuthToken } from './storeAuthToken.js';
