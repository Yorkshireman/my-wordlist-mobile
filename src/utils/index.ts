export { boldenSentenceForm } from './boldenSentenceForm';
export { calculateLongestWordLength } from './calculateLongestWordLength';
export { customColours } from './customColours';
export { displayLanguage } from './displayLanguage';
export { isInvalidEmail } from './isInvalidEmail';
export { parseCategories } from './parseCategories';
export { parseUniqueCategories } from './parseUniqueCategories';
export { sanitiseText } from './sanitiseText';
export { signIn } from './signIn';
export { signUp } from './signUp';
export { storeAuthToken } from './storeAuthToken';
