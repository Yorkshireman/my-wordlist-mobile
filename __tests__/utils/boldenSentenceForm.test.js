import { boldenSentenceForm } from '../../src/utils';

describe('boldenSentenceForm', () => {
  describe('when form is not present', () => {
    it('should return the sentence content wrapped in a <Text> component', () => {
      const sentence = {
        content: 'That was technologically sound.',
        id: '123'
      };

      expect(boldenSentenceForm(sentence)).toMatchSnapshot();
    });
  });

  describe('when form is present', () => {
    it('should return the sentence wrapped in a <Text> component with the form wrapped in a boldened <Text> component', () => {
      const sentence = {
        content: 'That was technologically sound.',
        form: 'technologically',
        id: '123'
      };

      expect(boldenSentenceForm(sentence)).toMatchSnapshot();
    });

    it('when content contains words that partially match the form, it should not bolden any part of that word', () => {
      const sentence = {
        content:
          'After showing his styler the style he wanted, he styled his hair stylishly until it was perfectly styled in the style he wanted.',
        form: 'style',
        id: '123'
      };

      expect(boldenSentenceForm(sentence)).toMatchSnapshot();
    });
  });
});
