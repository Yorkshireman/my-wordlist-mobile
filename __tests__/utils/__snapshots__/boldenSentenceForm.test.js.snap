// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boldenSentenceForm when form is not present should return the sentence content wrapped in a <Text> component 1`] = `
<ForwardRef(Text)>
  That was technologically sound.
</ForwardRef(Text)>
`;

exports[`boldenSentenceForm when form is present should return the sentence wrapped in a <Text> component with the form wrapped in a boldened <Text> component 1`] = `
<ForwardRef(Text)>
  That was 
  <ForwardRef(Text)
    style={
      {
        "fontWeight": "bold",
      }
    }
  >
    technologically
  </ForwardRef(Text)>
   sound.
</ForwardRef(Text)>
`;

exports[`boldenSentenceForm when form is present when content contains words that partially match the form, it should not bolden any part of that word 1`] = `
<ForwardRef(Text)>
  After showing his styler the 
  <ForwardRef(Text)
    style={
      {
        "fontWeight": "bold",
      }
    }
  >
    style
  </ForwardRef(Text)>
   he wanted, he styled his hair stylishly until it was perfectly styled in the 
  <ForwardRef(Text)
    style={
      {
        "fontWeight": "bold",
      }
    }
  >
    style
  </ForwardRef(Text)>
   he wanted.
</ForwardRef(Text)>
`;
